src/src/main/java/neonracer/gui/parser/LayoutParser.java

```diff
@@ -0,0 +1,214 @@
+package neonracer.gui.parser;
+
+import neonracer.gui.annotation.BindWidget;
+import neonracer.gui.annotation.EventHandler;
+import neonracer.gui.annotation.ParserMethod;
+import neonracer.gui.screen.Screen;
+import neonracer.gui.util.PropertyList;
+import neonracer.gui.widget.base.Container;
+import neonracer.gui.widget.base.Widget;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+
+class LayoutParser {
+
+    private static final String ROOT_ELEMENT = "Screen";
+
+    private Document document;
+
+    private LayoutParser(Document document) {
+        this.document = document;
+    }
+
+    static LayoutParser fromStream(InputStream inputStream) {
+        if (inputStream == null) throw new LayoutParserException("Could not find the layout file");
```

Comment:
In Java verwendet man normale eine `IllegalArgumentException` für Argumente, die nicht `null` sein dürfen (siehe [StackOverflow](https://stackoverflow.com/questions/3881/illegalargumentexception-or-nullpointerexception-for-a-null-parameter))

Reply:
Hast recht.